muduo/base/FileUtil.cc

```diff
@@ -31,25 +31,24 @@ FileUtil::AppendFile::~AppendFile()
 
 void FileUtil::AppendFile::append(const char* logline, const size_t len)
 {
-  size_t n = write(logline, len);
-  size_t remain = len - n;
-  while (remain > 0)
+  size_t total = 0, remain = len;
```

Comment:
Please follow the pattern of https://github.com/chenshuo/recipes/blob/master/tpc/lib/TcpStream.cc#L47-L63 to implement the loop.  Namely, one state variable is better than two.
```
size_t written = 0;
while (written < len)
{
  const size_t remain = len - written;
  size_t nw = write(logline + total, remain);
  if (nw != remain)
  {
    // error handling
    break;
  }
  written += nw;
}
writtenBytes_ += written;
```

Thanks.

Reply:
ok